source/m2mbase.cpp

```diff
@@ -279,10 +266,8 @@ M2MBase::Observation M2MBase::observation_level() const
 void M2MBase::get_observation_token(uint8_t *&token, uint32_t &token_length)
 {
     token_length = 0;
-    if(token) {
-        free(token);
-        token = NULL;
-    }
+    free(token);
```

Comment:
Potential crash. What if token passed is NULL, calling free() might cause crash.